src/Momento.Sdk/Config/Middleware/FairAsyncSemaphore.cs

```diff
@@ -0,0 +1,39 @@
+﻿using System;
+using System.Threading.Channels;
+using System.Threading.Tasks;
+
+namespace Momento.Sdk.Config.Middleware
+{
+    public class FairAsyncSemaphore
```

Comment:
this class seems fine the only thing seems little off is this is in Middleware package? Seems like a class to help with general concurency? Just confused me for second thought this was some other middle ware till I realized it was consumed by rate limited one further down. Not big issue just calling out made me double take for second.